### PR TITLE
Fixing the strong imported async frame pointer flags

### PR DESCRIPTION
**Pull request by etcwilde**
The weakly-imported symbol was getting optimized out, then put back in
as a strongly-imported symbol. This is no good. The symbol is a
declaration though, so it can't be "used" directly. Instead, we assign
it to another global and "use" it. That avoids the optimizations and
should be fine. Even if that symbol is a nullptr because it doesn't
exist, we are taking the pointer to it, which should be fine for all
situations.

Fixes rdar://84877644.

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
@swift-ci please test

File: lib/IRGen/IRGenModule.cpp [hunk 1]
```diff
@@ -1834,6 +1834,19 @@ void IRGenModule::emitSwiftAsyncExtendedFrameInfoWeakRef() {
...
                                          llvm::GlobalValue::ExternalWeakLinkage, nullptr,
                                          symbolName);
+
+  // The weak imported extendedFramePointerFlagsWeakRef gets optimized out
+  // before being added back as a strong import.
+  // Declarations can't be added to the used list, so we create a little
+  // global that can't be used from the program, but can be in the used list to
+  // avoid optimizations.
+  llvm::GlobalVariable *usage = new llvm::GlobalVariable(
+      Module, extendedFramePointerFlagsWeakRef->getTye(), false,
```
Comment (DougGregor) [inline lib/IRGen/IRGenModule.cpp:1844]:
```suggestion
      Module, extendedFramePointerFlagsWeakRef->getType(), false,
```

Comment (etcwilde) [reply lib/IRGen/IRGenModule.cpp]: grr. Thanks

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
Linux failures are due to the compile module being created by an older version of the compiler.

**Comment by etcwilde** [comment]
@swift-ci please clean test Linux

**Comment by DougGregor** [comment]
@swift-ci please test macOS

**Comment by etcwilde** [comment]
The macOS failure is due to other symbols being included in the used list. I really just want to verify that `(i8*** @_swift_async_extendedFramePointerFlagsUser to i8*)` is in the list.

The full list on the bot is

```
appending global [10 x i8*] [i8* bitcast ({ i32 }* @"\01l_entry_point" to i8*),
i8* bitcast (void (%swift.context*)* @"$s4main1fyyYaF" to i8*),
i8* bitcast (%swift.async_func_pointer* @"$s4main1fyyYaFTu" to i8*),
i8* bitcast (void (%swift.context*)* @"$s4main1gSSyYaF" to i8*),
i8* bitcast (%swift.async_func_pointer* @"$s4main1gSSyYaFTu" to i8*),
i8* bitcast (i16* @__swift_reflection_version to i8*),
i8* bitcast (void ()** @"_swift_FORCE_LOAD_$_swiftCompatibility51_$_main" to i8*),
i8* bitcast (void ()** @"_swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_main" to i8*),
i8* bitcast (i8*** @_swift_async_extendedFramePointerFlagsUser to i8*),
i8* bitcast (i32 (i32, i8**)* @main to i8*)],
section "llvm.metadata"
```

compared to the following list that I had in the test

```
appending global [5 x i8*] [i8* bitcast ({ i32 }* @"\01l_entry_point" to i8*), 
i8* bitcast (i16* @__swift_reflection_version to i8*), 
i8* bitcast (void ()** @"_swift_FORCE_LOAD_$_swiftCompatibilityConcurrency_$_test" to i8*), 
i8* bitcast (i8*** @_swift_async_extendedFramePointerFlagsUser to i8*),
i8* bitcast (i32 (i32, i8**)* @main to i8*)],
section "llvm.metadata"
```

**Comment by etcwilde** [comment]
@swift-ci please test

**Review by aschwaighofer — APPROVED** [review]
LGTM

**Comment by etcwilde** [comment]
@swift-ci please clean test Linux

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
The macOS bot may be signal. I'm trying to reproduce the `lldb-api :: lang/swift/playgrounds/TestPlaygrounds.py` failure locally because the log doesn't give very much information. 

```
FAIL: LLDB (/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_cross_module_extension_dsym (TestPlaygrounds.TestSwiftPlaygrounds)
UNSUPPORTED: LLDB (/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_cross_module_extension_dwarf (TestPlaygrounds.TestSwiftPlaygrounds) (skipping due to the following parameter(s): debug info format) 
UNSUPPORTED: LLDB (/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_cross_module_extension_dwo (TestPlaygrounds.TestSwiftPlaygrounds) (test case does not fall in any category of interest for this run) 
UNSUPPORTED: LLDB (/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_cross_module_extension_gmodules (TestPlaygrounds.TestSwiftPlaygrounds) (skipping (gmodules only makes sense for clang tests)) 
======================================================================
FAIL: test_cross_module_extension_dsym (TestPlaygrounds.TestSwiftPlaygrounds)
   Test that playgrounds work
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 1846, in test_method
    return attrvalue(self)
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 150, in wrapper
    return func(*args, **kwargs)
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 150, in wrapper
    return func(*args, **kwargs)
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 150, in wrapper
    return func(*args, **kwargs)
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/test/API/lang/swift/playgrounds/TestPlaygrounds.py", line 79, in test_cross_module_extension
    self.do_test(True)
  File "/Users/buildnode/jenkins/workspace/swift-PR-macos/branch-main/llvm-project/lldb/test/API/lang/swift/playgrounds/TestPlaygrounds.py", line 150, in do_test
    self.assertTrue("=\\'23\\'" in playground_output)
AssertionError: False is not True
```

The only way I can see for that particular test to fail is if something fails to import.

**Comment by etcwilde** [comment]
https://github.com/apple/llvm-project/pull/3520

@swift-ci please test macOS

**Comment by etcwilde** [comment]
@swift-ci please test macOS

**Pull request merged**